x-pack/test/dataset_quality_api_integration/tests/data_streams/degraded_docs.spec.ts

```diff
@@ -112,5 +110,175 @@ export default function ApiTest({ getService }: FtrProviderContext) {
         expect(stats.body.degradedDocs.length).to.be(0);
       });
     });
+
+    describe('when there are data streams of different spaces', () => {
+      const spaces = ['default', 'space1', 'space2'];
+      const datasetsWithNoDegradedDocs = ['nginx.access', 'apache.access', 'mysql.access'];
+      const datasetsWithDegradedDocs = ['nginx.error', 'apache.error', 'mysql.error'];
+
+      before(async () => {
+        for (const space of spaces) {
+          for (const dataset of datasetsWithNoDegradedDocs) {
+            await synthtrace.index([
+              timerange(start, end)
+                .interval('1m')
+                .rate(1)
+                .generator((timestamp) =>
+                  log
+                    .create()
+                    .message('This is a log message')
+                    .timestamp(timestamp)
+                    .dataset(dataset)
+                    .namespace(space)
+                ),
+            ]);
+          }
+
+          for (const dataset of datasetsWithDegradedDocs) {
+            await synthtrace.index([
+              timerange(start, end)
+                .interval('1m')
+                .rate(2)
+                .generator((timestamp: number, index: number) =>
+                  log
+                    .create()
+                    .message('This is a log message')
+                    .timestamp(timestamp)
+                    .dataset(dataset)
+                    .namespace(space)
+                    .logLevel(index % 2 === 0 ? MORE_THAN_1024_CHARS : 'This is a log message')
```

Comment:
nit: Why do we need to selectively choose to have degraded log level. This whole indexing is about degradedDocs, so we can remove this conditional check

Reply:
This is to simulate and assert that there are 50% degraded docs.